docs/cpu_templates/cpu-templates.md

```diff
@@ -0,0 +1,168 @@
+# CPU templates
+
+Firecracker allows users to customise how the vCPUs are represented
+to the guest software by changing the following configuration:
+
+- CPUID (x86_64 only)
+- MSRs (Model Specific Registers, x86_64 only)
+- ARM registers (aarch64 only)
+
+A combination of the changes to the entities above is called a CPU template.
+
+The functionality can be used when a user wants to mask a feature from
+the guest. One of real world use cases for this is representing a heterogeneous
+fleet (a fleet consisting of multiple CPU models) as a homogeneous fleet,
+so the guests will experience a consistent feature set supported by the host.
+
+Note: representing one CPU vendor as another CPU vendor is not supported.
+
+Firecracker supports two types of CPU templates:
+
+- Static CPU templates - a set of preconfigured CPU templates for users
+  to choose from;
```

Comment:
maybe better: "... a set of build in CPU templates ...."?

Reply:
indeed, thanks.